larevt/CalibrationDBI/Providers/DBFolder.cxx

```diff
@@ -229,81 +255,639 @@ namespace lariov {
+		 << "&t=" << ts.DBStamp();
+	if (fTag.length() > 0) fullurl2 << "&tag=" << fTag;
+	std::cout << "Full url = " << fullurl2.str() << std::endl;
+	int err = 0;
+	Dataset data = getDataWithTimeout(fullurl2.str().c_str(), NULL, fMaximumTimeout, &err);
+	int status = getHTTPstatus(data);
+	if (status != 200) {
+	  std::string msg = "HTTP error from " + fullurl2.str()+": status: " + std::to_string(status) + ": " + std::string(getHTTPmessage(data));
+	  throw WebError(msg);
+	}
+	DBDataset compare2(data, true);
+	CompareDataset(fCache, compare2);
+      }
     }
+    return true;
+  }
+
+  // Query data from sqlite database.
+  // The return value of type Dataset (aka void*), is partially opaque type HttpResponse*
+  // (defined in wda.c and copied above).
 
-    //update info about cached data
-    fNRows = getNtuples(fCachedDataset) - kNUMBER_HEADER_ROWS;
-    //std::cout<<"Retrieved "<<fNRows<<" rows from "<<fullurl.str()<<std::endl;
-    if (fNRows < 1) {
-      std::stringstream msg;
-      msg << "Time " << ts.DBStamp() << ": Data not found in database.";
-      throw WebError(msg.str());
-      fCachedStart = fCachedEnd = ts;
+  void DBFolder::GetSQLiteData(int t, DBDataset& data) const
+  {
+    if(fSQLitePath == "")
+      return;
+
+    //std::cout << "DBFolder::GetSQLiteData" << std::endl;
+    //std::cout << "t=" << t << std::endl;
+    //std::cout << "sqlite path = " << fSQLitePath << std::endl;
+
+    // Open sqlite database.
+
+    //std::cout << "Opening sqlite database " << fSQLitePath << std::endl;
+    sqlite3* db;
+    int rc = sqlite3_open(fSQLitePath.c_str(), &db);
+    if(rc != SQLITE_OK) {
+      std::cout << "Failed to open sqlite database " << fSQLitePath << std::endl;
+      throw cet::exception("DBFolder") << "Failed to open sqlite database " << fSQLitePath;
     }
 
-    //start and end times
-    Tuple tup;
-    tup = getTuple(fCachedDataset, 0);
-    char buf[kBUFFER_SIZE];
-    getStringValue(tup,0, buf, kBUFFER_SIZE, &err);
-    fCachedStart = IOVTimeStamp::GetFromString(std::string(buf));
-    releaseTuple(tup);
+    // Query begin time of IOV.
+
+    std::string table_iovs = fFolderName + "_iovs";
+    std::string table_tag_iovs = fFolderName + "_tag_iovs";
+    std::ostringstream sql;
+    sql << "SELECT " << table_iovs << ".iov_id," << table_iovs << ".begin_time"
+	<< " FROM " << table_tag_iovs << "," << table_iovs
+	<< " WHERE " << table_tag_iovs << ".tag='" << fTag << "'"
+	<< " AND " << table_tag_iovs << ".iov_id=" << table_iovs << ".iov_id"
+	<< " AND " << table_iovs << ".begin_time <= " << t
+	<< " ORDER BY " << table_iovs << ".begin_time desc";
+    //std::cout << "sql = " << sql.str() << std::endl;
+
+    // Prepare query.
+
+    sqlite3_stmt* stmt;
+    rc = sqlite3_prepare_v2(db, sql.str().c_str(), -1, &stmt, 0);
+    if(rc != SQLITE_OK) {
+      std::cout << "sqlite3_prepare_v2 failed." << fSQLitePath << std::endl;
+      std::cout << "Failed sql = " << sql.str() << std::endl;
```

Comment:
Suggest removing `std::cout` statements just before exception throw--include the relevant message in the exception.